### PR TITLE
Add Model API as a full-fledged library

### PR DESCRIPTION
**Pull request by atobiszei**
* Updated bazel rules_foreign to commit past 0.9
This is required to be able to use cmake >= 3.262 required for Model API

JIRA:CVS-120866

### CODE REVIEW

File: third_party/model_api/BUILD [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+)
+
+cmake(
+    name = "model_api_cmake",
+    cache_entries = {
+        "CMAKE_POSITION_INDEPENDENT_CODE": "ON",
+        "OpenVINO_DIR": "/opt/intel/openvino/runtime/cmake",
+    },
+    env = {
+        "HTTP_PROXY": "http://proxy-dmz.intel.com:911",
```
Comment (dtrawins) [inline third_party/model_api/BUILD]: don't include proxy

File: third_party/model_api/BUILD [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    deps = [
+        "@mediapipe//mediapipe/framework/port:opencv_core",
+        "@linux_openvino//:openvino",
+        ":model_api_cmake",
+    ],
+    visibility = ["//visibility:public"],
+)
+
+cc_library(
+   name = "adapter_api",
```
Comment (mgumowsk) [inline third_party/model_api/BUILD]: There's no need for separate "adapter_api". I would suggest to remove it, and it now ovms calculators use "@model_api//:model_api"

**Review by mgumowsk — CHANGES_REQUESTED** [review]

File: WORKSPACE [hunk 3]
```diff
@@ -71,16 +71,24 @@ http_archive(
     urls = ["https://github.com/bazelbuild/rules_cc/archive/2f8c04c04462ab83c545ab14c0da68c3b4c96191.zip"],
 )
 
-http_archive(
+#http_archive(
+#    name = "rules_foreign_cc",
```
Comment (rasapala) [inline WORKSPACE]: Remove

**Review by rasapala — APPROVED** [review]

**Review by mgumowsk — APPROVED** [review]

File: third_party/model_api/BUILD (hunk 1 above)
Comment (mgumowsk) [reply third_party/model_api/BUILD]: I believe proxy is required by cmake to download dependencies.

**Pull request merged**